internal/controllers/nft_controller.go

```diff
@@ -215,12 +200,7 @@ func (nc *NFTController) GetNFTImage(c *fiber.Ctx) error {
 		return opaqueInternalError
 	}
 
-	if nft.R.UserDevice == nil {
-		helpers.SkipErrorLog(c)
-		return fiber.NewError(fiber.StatusNotFound, "NFT not found.")
-	}
-
-	imageName = nft.MintRequestID
+	imageName := nft.MintRequestID.String
```

Comment:
The old code assumes that `mint_request_id` is always populated, since that's the primary key of the table; this is no longer a safe assumption. And in fact right now we're not able to provide an image for vehicles whose mint calls were directly to the chain.

I think if `mint_request_id` is null then we should error here.

Reply:
Fixed it, I think.